pkg/controller/bootstrap/tokencleaner.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/api"
+	apierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/metrics"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+// TokenCleanerOptions contains options for the TokenCleaner
+type TokenCleanerOptions struct {
+
+	// SecretResync is the time.Duration at which to fully re-list secrets.
+	// If zero, re-list will be delayed as long as possible
+	SecretResync time.Duration
+}
+
+// DefaultTokenCleanerOptions returns a set of default options for creating a
+// TokenCleaner
+func DefaultTokenCleanerOptions() TokenCleanerOptions {
+	return TokenCleanerOptions{}
+}
+
+// TokenCleaner is a controller that deletes expired tokens
+type TokenCleaner struct {
+	stopChan chan struct{}
+
+	client clientset.Interface
+
+	secrets           cache.Store
+	secretsController *cache.Controller
+}
+
+func NewTokenCleaner(cl clientset.Interface, options TokenCleanerOptions) *TokenCleaner {
+	e := &TokenCleaner{
+		client: cl,
+	}
+	if cl != nil && cl.Core().RESTClient().GetRateLimiter() != nil {
+		metrics.RegisterMetricAndTrackRateLimiterUsage("token_cleaner", cl.Core().RESTClient().GetRateLimiter())
+	}
+
+	secretSelector := fields.SelectorFromSet(map[string]string{api.SecretTypeField: string(api.SecretTypeBootstrapToken)})
+	e.secrets, e.secretsController = cache.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(lo v1.ListOptions) (runtime.Object, error) {
+				lo.FieldSelector = secretSelector.String()
+				return e.client.Core().Secrets(api.NamespaceSystem).List(lo)
+			},
+			WatchFunc: func(lo v1.ListOptions) (watch.Interface, error) {
+				lo.FieldSelector = secretSelector.String()
+				return e.client.Core().Secrets(api.NamespaceSystem).Watch(lo)
+			},
+		},
+		&v1.Secret{},
+		options.SecretResync,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    e.evalSecret,
+			UpdateFunc: func(old, new interface{}) { e.evalSecret(new) },
```

Comment:
Maybe `func(oldSecret, newSecret interface{}) { e.evalSecret(newSecret) }`?
I first interpreted `new` as a keyword and wondered if something was missing until I saw that new was passed as an argument...